File: sdks/csharp/src/NovaBilling/Types/CustomerAnalyticsResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record CustomerAnalyticsResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("totalCustomers")]
    public required double TotalCustomers { get; set; }

    [JsonPropertyName("newCustomers")]
    public required double NewCustomers { get; set; }

    /// <summary>
    /// Average revenue per user
    /// </summary>
    [JsonPropertyName("arpu")]
    public required string Arpu { get; set; }

    [JsonPropertyName("totalRevenue")]
    public required string TotalRevenue { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/ruby/lib/novabilling/types/customer_analytics_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class CustomerAnalyticsResponse < Internal::Types::Model
      field :total_customers, -> { Integer }, optional: false, nullable: false, api_name: "totalCustomers"
      field :new_customers, -> { Integer }, optional: false, nullable: false, api_name: "newCustomers"
      field :arpu, -> { String }, optional: false, nullable: false
      field :total_revenue, -> { String }, optional: false, nullable: false, api_name: "totalRevenue"
    end
  end
end
